Makefile

```
@@ -442,6 +459,9 @@ install: $(wildcard *.go)
 .PHONY: build
 build: go-check generate $(EXECUTABLE)
 
+.PHONY: build-offline
+build-offline: build
```

What about something like?:

```suggestion
generate: fomantic webpack $(TAGS_PREREQ)
    $(MAKE) do-generate

do-generate:
    GO111MODULE=on $(GO) generate -mod=vendor ...

build-offline:  do-generate $(EXECUTABLE)
```
I didn't check the syntax, but the idea is to avoid executing anything  related to fomantic or webpack in the offline-build, as everything is already packed in the tar file if I understood correctly.

This should simplify things a little, I hope.

---

I think even offline build should actually "build" and not ship prebuilt files so I'd lean the other way and actually exclude the webpack files in the source tarball.

---

Actually, these offline builds are all about repeatability. `make` itself makes no guarantees about the order in which the dependencies are evaluated. Obviously different versions of npm and webpack can produce different outputs, but let's say the npm and webpack versions are freezed: can they still produce non-deterministic outputs due to parallelization?

---

Our complete dependency tree is frozen, the only non-determinism (besides module bugs) can come from platform-specific behaviour (for example `fsevents` dependency is only used on macOS but as I understand it, it's just a performance thing).

Given how complex a webpack build is in general, I'd certainly assume there are a few module bugs that may introduce differences in the output based on various factors, like the Node.js version it runs on etc.

Thought I'd say even if there are issues like I think we better remove all built files and caches from the source tarball which would be at least:

- `public/js`
- `public/css`
- `public/img/svg`
- `node_modules/.cache`
- `node_modules/**/*.node` (native node modules may be non-portable)

---

Implemented all excludes except the `node_modules/**/*.node` one. It's a non-issue as long as release on any platform besides macOS as [fsevents only builds there](https://github.com/fsevents/fsevents/blob/9cdfb441a818ad3ba18b0197a0799a0e9edec49f/package.json#L8).

---

Regarding the original request to exclude frontend targets: I think it's better to keep them in so one can `make clean-all build-offline` which would otherwise not produce any webpack assets. Would you agree @guillep2k

---

I'm not sure what the use case would be, but I have no objections. 👍